### ISSUE TITLE
F401 false positive for .pyi files

### ISSUE THREAD

**Issue by JonathanPlasse** [charliermarsh/ruff#3014]
```python
import queue

class Serial:
    queue: queue.Queue[bytes | None] | None
```

`ruff` detect an `F401` error while `flake8` does not.
It works for `.py` files but not for `.pyi`.

- I did not add a playground link as I could not reproduce the error with it (c.f. #3013).

**Comment by ngnpope** [comment]
Duplicate of #1401.
